sloghandler.go

```diff
@@ -0,0 +1,71 @@
+//go:build go1.21
+// +build go1.21
+
+/*
+Copyright 2023 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logr
+
+import (
+	"context"
+	"log/slog"
+)
+
+type slogHandler struct {
+	sink        LogSink
+	groupPrefix string
+}
+
+func (l *slogHandler) Enabled(ctx context.Context, level slog.Level) bool {
+	return l.sink.Enabled(levelFromSlog(level))
+}
+
+func (l *slogHandler) Handle(ctx context.Context, record slog.Record) error {
```

Comment:
We need to document somewhere that the passed-in timestamp is ignored, and any `logr.LogSink` which logs the time is going to produce its own timestamp.

Reply:
The README has a list of drawbacks that come from using glue code for "logr on slog" - this is another one which I need to add.